snekbox/nsjail.py

```diff
@@ -133,11 +136,33 @@ def python3(self, code: str) -> CompletedProcess:
             except ValueError:
                 return CompletedProcess(args, None, "ValueError: embedded null byte", None)
 
+            output_size = 0
+            output = []
+
+            # We'll consume STDOUT as long as the subprocess is running
+            while nsjail.poll() is None:
+                # Read 100 characters from the STDOUT stream
+                chars = nsjail.stdout.read(100)
```

Comment:
100 seems really low. I suggest something in the 10's of kilobytes. Since an overflow of that magnitude is still insignificant, it should not discard the overflowed bytes it reads.

Reply:
Consider using `io.DEFAULT_BUFFER_SIZE` or setting a custom `bufsize` for `Popen`.

Reply:
I can increase the number of characters read from the stream, although this increases the uncertainty about the size in bytes we're reading due to UTF-8 characters being up to 4 bytes long.

Reply:
> Consider using `io.DEFAULT_BUFFER_SIZE` or setting a custom `bufsize` for `Popen`.

This is the default value for Popen due to the negative default value for bufsize (`-1`):

> negative bufsize (the default) means the system default of io.DEFAULT_BUFFER_SIZE will be used.

Reply:
> this increases the uncertainty about the size in bytes we're reading due to UTF-8 characters being up to 4 bytes long.

Well, even being 40 KB over is not a big deal.

> This is the default value for Popen due to the negative default value for bufsize

That was worded poorly. I mean that when you read from the stream, use `read(io.DEFAULT_BUFFER_SIZE)` or set the `bufsize` to the same number you use to read from the stream. Not sure if it really makes a difference.